torchtext/experimental/datasets/raw/amazonreviewpolarity.py

```diff
@@ -0,0 +1,39 @@
+from torchtext.utils import download_from_url, extract_archive, unicode_csv_reader
+from torchtext.experimental.datasets.raw.common import RawTextIterableDataset
+from torchtext.experimental.datasets.raw.common import wrap_split_argument
+from torchtext.experimental.datasets.raw.common import add_docstring_header
+from torchtext.experimental.datasets.raw.common import find_match
+import os
+import io
+
+URL = 'https://drive.google.com/uc?export=download&id=0Bz8a_Dbh9QhbaW12WVVZS2drcnM'
+
+MD5 = 'fe39f8b653cada45afd5792e0f0e8f9b'
+
+NUM_LINES = {
+    'train': 3600000,
+    'test': 400000,
+}
+
+_PATH = 'amazon_review_polarity_csv.tar.gz'
+
+
+@wrap_split_argument
+@add_docstring_header()
+def AmazonReviewPolarity(root='.data', split=('train', 'test'), offset=0):
+    def _create_data_from_csv(data_path):
```

Comment:
I feel we don't need to repeat `_create_data_from_csv` function for all the datasets. We can have a utility func and keep it private somewhere.

Reply:
Yes, I agree. I'll think of some generic form that we can share across all datasets, which is another step after the datasets have been split.